apps/mymedicare_cb/views.py

```diff
@@ -60,48 +69,98 @@ def authenticate(request):
-    sls_mbi_format_synthetic = is_mbi_format_synthetic(sls_mbi)
+    # Set identity values from userinfo response.
+    sls_subject = user_info.get("sub", None).strip()
+    sls_hicn = user_info.get("hicn", "").strip()
+    #     Convert SLS's mbi to UPPER case.
+    sls_mbi = user_info.get("mbi", "").strip().upper()
+    sls_first_name = user_info.get('given_name', "")
+    sls_last_name = user_info.get('family_name', "")
+    sls_email = user_info.get('email', "")
 
     # If MBI returned from SLS is blank, set to None for hash logging
     if sls_mbi == "":
         sls_mbi = None
 
-    authenticate_logger.info({
-        "type": "Authentication:start",
-        "sub": user_info["sub"],
-        "sls_mbi_format_valid": sls_mbi_format_valid,
-        "sls_mbi_format_msg": sls_mbi_format_msg,
-        "sls_mbi_format_synthetic": sls_mbi_format_synthetic,
-        "sls_hicn_hash": hash_hicn(user_info['hicn']),
-        "sls_mbi_hash": hash_mbi(sls_mbi),
-    })
-
-    user = get_and_update_user(user_info)
-
-    authenticate_logger.info({
-        "type": "Authentication:success",
-        "sub": user_info["sub"],
-        "user": {
-            "id": user.id,
-            "username": user.username,
-            "crosswalk": {
-                "id": user.crosswalk.id,
-                "user_hicn_hash": user.crosswalk.user_hicn_hash,
-                "user_mbi_hash": user.crosswalk.user_mbi_hash,
-                "fhir_id": user.crosswalk.fhir_id,
-                "user_id_type": user.crosswalk.user_id_type,
-            },
-        },
-    })
+    # Validate: sls_subject cannot be empty. TODO: Validate format too.
+    if sls_subject == "":
+        mesg = "User info sub cannot be empty"
+        log_authenticate_start(authenticate_logger, "FAIL", mesg)
+        raise UpstreamServerException(
+            "An error occurred validating identity information from provider." + mesg)
+
+    # Validate: sls_hicn cannot be empty.
+    if sls_hicn == "":
+        mesg = "User info HICN cannot be empty."
+        log_authenticate_start(authenticate_logger, "FAIL", mesg, sls_subject)
+        raise UpstreamServerException(
+            "An error occurred validating identity information from provider." + mesg)
+
+    # Set Hash values once here for performance and logging.
+    sls_hicn_hash = hash_hicn(sls_hicn)
+    sls_mbi_hash = hash_mbi(sls_mbi)
+
+    # Validate: hicn_hash expected length = 64.
+    if len(sls_hicn_hash) != 64:
+        mesg = "Incorrect HICN hash format length of " + str(len(sls_hicn_hash)) + " is != 64 ."
+        log_authenticate_start(authenticate_logger, "FAIL", mesg,
+                               sls_subject, None,
+                               None, None,
+                               sls_hicn_hash, sls_mbi_hash)
+        raise UpstreamServerException(
+            "An error occurred calculating the ID hash." + mesg)
+
+    # Validate: mbi_hash expected length = 64.
+    if len(sls_mbi_hash) != 64:
```

Comment:
I was testing this locally and noticed this will break if SLS (mSLS in my case) does not return a MBI as you can't get the len() of a None type.  Should this be put within the `if sls_mbi is not None:` statement below instead to validate the MBI hash?

Reply:
@njdister I just pushed up changes to fix that. TY for catching that. I didn't test that out after refactoring. Had some other issues to fix related to that too.